File: src/030. Digit fifth powers/030.go
package main

import "fmt"

func main() {
	sum := 0
	for i:= 2; i < 1000000; i++ {
		if digitFifthPowerSum(i) == i {
			sum += i
		}
	}
	fmt.Println(sum)
}

func digitFifthPowerSum(n int) int {
	sum := 0
	for n > 0 {
		d := n % 10
		sum += d * d * d * d * d
		n /= 10
	}
	return sum
}

File: src/030. Digit fifth powers/030.java
public class Main {
    
    public static void main(String[] args) {

        int sum = 0;
        for (int i = 2; i < 1000000; i++) {
            if (digitFifthPowerSum(i) == i) {
                sum += i;
            }
        }
        System.out.println(sum);
    }

    private static int digitFifthPowerSum(int n) {
        int sum = 0;
        while (n > 0) {
            int d = n % 10;
            sum += d * d * d * d * d;
            n /= 10;
        }
        return sum;
    }

}
